packages/authorization-updater/src/index.ts

```diff
@@ -332,10 +340,110 @@ export async function sendPurposeAuthUpdate(
     .exhaustive();
 }
 
+export async function sendAuthorizationAuthUpdate(
+  decodedMessage: AuthorizationEventV2,
+  authService: AuthorizationService,
+  readModelService: ReadModelService,
+  logger: Logger,
+  correlationId: string
+): Promise<void> {
+  await match(decodedMessage)
+    .with({ type: "ClientAdded" }, async (msg): Promise<void> => {
+      const client = getClientFromEvent(msg, msg.type);
+
+      if (!client) {
+        throw missingKafkaMessageDataError("client", msg.type);
+      }
+      await authService.addClient(client, logger, correlationId);
+    })
+    .with({ type: "ClientDeleted" }, async (msg): Promise<void> => {
+      const clientId = unsafeBrandId<ClientId>(msg.data.clientId);
+
+      if (!clientId) {
+        throw missingKafkaMessageDataError("client", msg.type);
+      }
+      await authService.deleteClient(clientId, logger, correlationId);
+    })
+    .with({ type: "ClientKeyAdded" }, async (msg) => {
+      if (!msg.data.client) {
+        throw missingKafkaMessageDataError("client", msg.type);
+      }
+      const client = fromClientV2(msg.data.client);
+      const clientId = client.id;
+      const kid = msg.data.kid;
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const key = client.keys.find((key) => key.kid === kid)!;
```

Comment:
why not a check like [this](https://github.com/pagopa/interop-be-monorepo/pull/675/files?diff=split&w=1#diff-5d1776c097c6d2b33d3576fddb1f6c61aa5522c6d061deab2aa9859630e7fa30R362)?

Reply:
Updated